datastore/keycompat.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2019 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+package datastore
+
+import (
+	"sync"
+
+	"golang.org/x/net/context"
+
+	"google.golang.org/appengine/datastore/internal/cloudkey"
+	"google.golang.org/appengine/internal"
+)
+
+var keyConversion struct {
+	mu      sync.RWMutex
+	project string
+}
+
+// EnableKeyConversion enables encoded key compatibility with the Cloud Datastore client library (cloud.google.com/go/datastore).
+// Encoded keys generated by the Cloud Datastore client library will be decoded into App Engine datastore keys.
+//
+// The context provided must be an App Engine context if running in first generation App Engine.
+// This can be called in the /_ah/start handler. It is safe to call multiple times, and is cheap to call, so can also be inserted as middleware.
+//
+// Enabling key compatibility not affect the encoding format used by Key.Encode, it only expands the type of keys that are able to be decoded with DecodeKey.
+func EnableKeyConversion(ctx context.Context) {
+	if getKeyConversionAppID() == "" {
```

Comment:
Am I misreading this? Won't `getKeyConversionAppID()` always return "" until the code below this runs? And so this effectively prevents `keyConversion.project` from ever getting set?

Reply:
Oops. Expression was inverted. PTAL. Good catch.